Assets/Scripts/AbstractClasses/Movable.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
+[RequireComponent(typeof(NavMeshAgent))]
+public abstract class Movable : MonoBehaviour
+{
+    [Tooltip("The speed to rotate with")]
+    [SerializeField]
+    private int rotationSpeed;
+
+    [Tooltip("The maximum difference in degrees between look direction and target direction in order to be facing the target.")]
+    [SerializeField]
+    private int rotationTolerance;
+
+
+    protected NavMeshAgent _navMeshAgent;
```

Comment:
```suggestion
    protected NavMeshAgent NavMeshAgent;
```
Naming conventions ;)

Reply:
didn't know a protected var (which is like private for each child) should be written like that